astropy/units/quantity.py

```diff
@@ -1081,21 +1081,23 @@ def __ilshift__(self, other):
         try:
             other = Unit(other, parse_strict='silent')
         except UnitTypeError:
-            return NotImplemented
+            return NotImplemented  # try other.__rlshift__(self)
 
         try:
             factor = self.unit._to(other)
-        except Exception:
-            # Maybe via equivalencies?  Now we do make a temporary copy.
-            try:
-                value = self._to_value(other)
-            except UnitConversionError:
-                return NotImplemented
-
-            self.view(np.ndarray)[...] = value
+        except UnitConversionError:  # incompatible, or requires an Equivalency
+            return NotImplemented
+        except AttributeError:  # StructuredUnit does not have `_to`
+            # In principle, in-place might be possible.
+            return NotImplemented
 
-        else:
-            self.view(np.ndarray)[...] *= factor
+        view = self.view(np.ndarray)
+        try:
+            view[...] *= factor  # operates on view
```

Comment:
`[...]` should not be needed - does something go wrong without it?

Reply:
I think I pushed the commit before GH loaded your reply. Undoing...